packages/ui/src/forum/types/ForumCategory.ts

```diff
@@ -47,12 +49,33 @@ export const asBaseForumCategory = (fields: ForumCategoryFields): Omit<ForumCate
   moderators: fields.moderators?.map(({ id, membership }) => ({ id, handle: membership.handle })) ?? [],
 })
 
+export const asBaseArchivedForumCategory = (
+  fields: ArchivedForumCategoryFields
+): Omit<ForumCategory, 'subcategories'> => ({
+  id: fields.id,
+  title: fields.title,
+  description: fields.description,
+  status: {
+    __typename: fields.status.__typename,
+    ...('categoryArchivalStatusUpdatedEvent' in fields.status && fields.status.categoryArchivalStatusUpdatedEvent
+      ? { categoryArchivalStatusUpdatedEvent: asBlock(fields.status.categoryArchivalStatusUpdatedEvent) }
+      : {}),
+  },
+  moderators: [],
+})
+
 export const asForumCategory = (fields: ForumCategoryFields): ForumCategory => ({
   ...asBaseForumCategory(fields),
   subcategories:
     fields.forumcategoryparent?.map((fields) => ({ ...asSubCategory(fields), status: fields.status.__typename })) ?? [],
 })
 
+export const asArchivedForumCategory = (fields: ArchivedForumCategoryFields): ForumCategory => ({
+  ...asBaseArchivedForumCategory(fields),
```

Comment:
Changing types in GraphQL queries would probably enforce changing some casting here as well.

I'd expect that `asBaseArchivedForumCategory()` would not be needed (the `asBaseForumCategory()` should be enough) and the `moderators` casting will probably be moved to `asForumCategory()` casting.

Reply:
Thanks for the tips, I applied the proposed changes.